### PR TITLE
spaces in keywords

### PR DESCRIPTION
**Pull request by SimonLab**
resolve the [question/issue 13](https://github.com/dwyl/search-result-keyword-highlighter/issues/13)

### CODE REVIEW

**Comment by nelsonic** [comment]
@SimonLab is this ready to be merged?
(**_Assign**_ _if ready_ :wink: )

File: lib/keyword-highlighter.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ function wrapper (match) {
  * @returns {String} text - the text with all highilights.
  */
 function highlight(keywords, text) {
-  var matcher = new RegExp(keywords.split(' ').join('|'), 'gi');
+  var matcher = new RegExp(keywords.split(' ').filter(k => k !== '').join('|'), 'gi');
```
Comment (nelsonic) [inline lib/keyword-highlighter.js]: remember that **_arrow functions**_ are **_not supported**_ in Safari and Internet Explorer ... https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions#Browser_compatibility
js-arrow-functions-not-supported-in-safari-or-internet-explorer

**Comment by SimonLab** [comment]
@nelsonic sorry I forget this PR I will update it soon

File: lib/keyword-highlighter.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ function wrapper (match) {
 }
 
 /**
- * highilight does Simple String Substitution given a set of keywords in 
+ * highilight does Simple String Substitution given a set of keywords in
```
Comment (varjmes) [inline lib/keyword-highlighter.js]: The word 'highilight' should be 'highlight', if you wanted to fix that, too :)

Comment (nelsonic) [inline lib/keyword-highlighter.js]: Well spotted @Charlotteis feel free to correct on this branch. :+1:

Comment (varjmes) [inline lib/keyword-highlighter.js]: > Well spotted @Charlotteis feel free to correct on this branch.

Thanks! Done :)

**Comment by SimonLab** [comment]
@nelsonic this PR is ready to be reviewed (and merged?). Thanks.

File: lib/keyword-highlighter.js [hunk 3]
```diff
@@ -19,17 +19,24 @@ function wrapper (match) {
...
  */
 function highlight(keywords, text) {
-  var regex = keywords.split(' ').join('|');
+
+  var regex = keywords.split(' ');
+  regex = regex.filter(function(char){
+    return char !== '';
+  });
+  regex = regex.join('|');
   regex = regex.replace(/[-[\]{}()*+?.,\\^$]/g, "\\$&");
```
Comment (nelsonic) [inline lib/keyword-highlighter.js:37]: @SimonLab please add comments to your regex next time it helps others understand the code...

**Comment by nelsonic** [comment]
looks good. merging. thanks! :+1:

**Comment by iteles** [comment]
Thanks @charlotteis and @SimonLab!

**Pull request merged**